docs/development/release.md

```diff
@@ -3,56 +3,75 @@
+    b. Release `rc0` images using the [release-image-build](https://github.com/ray-project/kuberay/actions/workflows/image-release.yaml) workflow on GitHub actions.
+    You will prompted for a commit reference and an image tag. The commit reference should be the SHA of the tip of the release branch. The image tag should be `vX.Y.Z-rc.0`.
+    c. Tag the tip of release branch with `vX.Y.Z-rc.0`.
     ```
-    git tag v0.2.0
-    git push upstream v0.2.0
+    git tag v0.4.0-rc.0
+    git push upstream v0.4.0-rc.0
     ```
+    d. Release rc0 Helm charts following the [instructions](../release/helm-chart.md).
+    e. Open a PR into the Ray repo updating the operator version used in the autoscaler integration test. Make any adjustments necessary for the test to pass. [Example](https://github.com/ray-project/ray/pull/30944/files). Make sure the test labelled [kubernetes-operator](https://buildkite.com/ray-project/oss-ci-build-pr/builds/7141#0184ef25-e62c-4dab-9c7e-ddfd583803cd) passes before merging.
+    f. Announce the `rc0` release on the KubeRay slack, with deployment instructions. [Example.](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1670375020308739).
+
+5. If necessary, create more release candidates (`rc1`, `rc2`, ...)
+    - Resolve issues with the release branch by cherry picking master commits
+into the release branch.
+    - When cherry-picking changes, it is best to open a PR against the release branch -- don't push directly to the release branch.
+    - When the next release candidate is ready, repeat step 4 above.
 
-1. Run following code and fetch online git commits from last release (v0.1.0) to current release (v0.2.0).
+6. Create a final release by repeating Step 4 once more using the tag of the release (`vX.Y.Z`) with no `-rc` suffix.
+
+7. Ask @Jeffwan to build and upload CLI binaries for the release.
+
+8. Run following code and fetch online git commits from last release (v0.3.0) to current release (v0.4.0).
 
     ```
-     git log v0.1.0..v0.2.0 --oneline
+    git log v0.3.0..v0.4.0 --oneline
     ```
- 
-1. Generate release notes and update Github release. See v0.1.0 example [here](https://github.com/ray-project/kuberay/releases/tag/v0.1.0). Please also upload CLI binaries.
 
-1. Send a PR to update [CHANGELOG.md](../../CHANGELOG.md)
- 
\ No newline at end of file
+9. Generate release notes and update the GitHub release. See v0.3.0 the example [here](https://github.com/ray-project/kuberay/releases/tag/v0.3.0).
+
+10. Send a PR to add the release notes to [CHANGELOG.md](../../CHANGELOG.md).
+
+11. Merge Ray and KubeRay PRs updating each occurence of the latest KubeRay version in the documentation.
```

Comment:
Do you mean updating 0.3.0 to 0.4.0 in related documents?

Reply:
Yes.

Reply:
I've rewritten it a bit -- it's now Step 10 -- let me know if the wording makes sense.